jbpm-services/jbpm-executor/src/main/java/org/jbpm/executor/impl/jpa/ExecutorQueryServiceImpl.java

```diff
@@ -228,15 +228,23 @@ public RequestInfo getRequestForProcessing(Long requestId) {
 		@Override
 		public RequestInfo execute(Context context) {
 			Map<String, Object> params = new HashMap<String, Object>();
-	    	params.put("now", new Date());
-	    	params.put("firstResult", 0);
-	    	params.put("maxResults", 1);
+	    	params.put("now", new Date());	    	
 	    	params.put("owner", ExecutorService.EXECUTOR_ID);
 	    	RequestInfo request = null;
 	    	try {
 	    		org.jbpm.shared.services.impl.JpaPersistenceContext ctx = (org.jbpm.shared.services.impl.JpaPersistenceContext) context;
-				request = ctx.queryAndLockWithParametersInTransaction("PendingRequestsForProcessing",params, true, RequestInfo.class);
-				
+	    		
+	    		if ("org.jbpm.persistence.jpa.hibernate.DisabledFollowOnLockOracle10gDialect".equals(ctx.get("hibernate.dialect"))) {
+	    		    
+	    		    Number lockedRequest = ctx.nativeQueryAndLockWithParametersInTransaction("NativePendingRequestsForProcessing",params, true, Number.class);
+	    		    if (lockedRequest != null) {
+	    		        request = ctx.find(org.jbpm.executor.entities.RequestInfo.class, lockedRequest.longValue());
+	    		    }
+	    		} else {
+	    		    params.put("firstResult", 0);
+	                params.put("maxResults", 1);
```

Comment:
Weird formatting, I would correct it, but it's up to you, not a big deal.

Reply:
same small formatting on line 239 in method params